src/EditorFeatures/Core/Shared/Extensions/DocumentExtensions.cs

```diff
@@ -8,10 +8,10 @@ namespace Microsoft.CodeAnalysis.Editor.Shared.Extensions
 {
     internal static class DocumentExtensions
     {
-        public static bool IsInCloudEnvironmentClientContext(this Document document)
+        public static bool IsInLspEditorContext(this Document? document)
         {
-            var workspaceContextService = document.Project.Solution.Workspace.Services.GetRequiredService<IWorkspaceContextService>();
-            return workspaceContextService.IsCloudEnvironmentClient();
+            var workspaceContextService = document?.Project.Solution.Workspace.Services.GetRequiredService<IWorkspaceContextService>();
+            return workspaceContextService?.IsInLspEditorContext() == true;
         }
```

Comment:
Can we remove this extension? It seems to be not used in many places. The reason is - Document is a snapshot and should have no state. It doesn't have any "context". The state is sneaking in thru Solution.Workspace, which we want to ultimately obsolete.

Reply:
Can do that as a follow up.

---
In reply to: [542629915](https://github.com/dotnet/roslyn/pull/49936#discussion_r542629915) [](ancestors = 542629915)

Reply:
Done!  Looks like all of the usages could use the text buffer extension method so swapped them all.